File: pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = (letters / words) * 100
    S = (sentences / words) * 100

    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 15:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    total = 0
    for i in range(len(text)):
        if text[i].isalpha():
            total += 1
    return total


def count_words(text):
    total = 1
    for i in range(len(text)):
        if text[i].isspace():
            total += 1
    return total


def count_sentences(text):
    total = 0
    for i in range(len(text)):
        if text[i].isalpha() == 0 and (text[i] == "." or text[i] == "!" or text[i] == "?"):
            total += 1
    return total


main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    float words = count_words(text);
    int sentences = count_sentences(text);

    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    int total = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            total++;
        }
    }
    return total;
}

int count_words(string text)
{
    int total = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            total++;
        }
    }
    return total;
}

int count_sentences(string text)
{
    int total = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]) == 0)
        {
            if (text[i] == 46 || text[i] ==  33 || text[i] == 63)
            {
                total++;
            }
        }
    }
    return total;
}